src/components/Pagination/Pagination.tsx

```diff
@@ -1,42 +1,142 @@
+import { ButtonTypes } from "../Button/ButtonTypes";
 import bem from "../../utils/bem";
+
 export interface PaginationProps {
     /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
     blockName?: string;
     /** Additional className for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
     className?: string;
     /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
     modifiers?: string[];
-    /** Right hand element that is the next handler, typically a button */
-    nextPage?: React.ReactNode;
-    /** Left hand element that is the previous handler, typically a button */
-    previousPage?: React.ReactNode;
+    /** The total number of pages */
+    pageCount: number;
+    /** The function to call when a page is clicked. */
+    onPageChange?: (selected) => void;
+}
+
+export interface PaginationState {
+    selected: number;
 }
 
-export default function Pagination(
-    props: React.PropsWithChildren<PaginationProps>
-) {
-    const {
-        blockName,
-        children,
-        className,
-        modifiers,
-        nextPage,
-        previousPage,
-    } = props;
-
-    const pagination__base_class = "pagination";
-
-    return (
-        <nav
-            aria-label="Pagination"
-            className={bem(pagination__base_class, modifiers, blockName, [
-                className,
-            ])}
-        >
-            {previousPage}
-            {children}
-            {nextPage}
-        </nav>
-    );
+/** Pagination component that provides a list of page items */
+export default class Pagination extends React.Component<
+    PaginationProps,
+    PaginationState
+> {
+    constructor(props: PaginationProps) {
+        super(props);
+        this.state = {
+            selected: 0,
+        };
+    }
+
+    previousPage = () => {
+        const { selected } = this.state;
+        if (selected > 0) {
+            this.selectPage(selected - 1);
+        }
+    };
+
+    nextPage = () => {
+        const { selected } = this.state;
+        const { pageCount } = this.props;
+
+        if (selected < pageCount - 1) {
+            this.selectPage(selected + 1);
+        }
+    };
+
+    selectPage = (index) => {
+        if (this.state.selected === index) return;
+
+        this.setState({ selected: index });
+
+        // Call the callback with the new selected item:
+        this.runCallback(index);
+    };
+
+    runCallback = (selectedItem) => {
+        if (
+            typeof this.props.onPageChange !== "undefined" &&
+            typeof this.props.onPageChange === "function"
+        ) {
+            this.props.onPageChange({ selected: selectedItem });
+        }
+    };
+
+    getPageElement(index) {
+        const { selected } = this.state;
+
+        return (
+            <li key={index + 1}>
+                <Button
```

Comment:
I think you want to be using `<Link>` here instead of `<Button>`, because the underlying native element you want is an `<a>` element rather than a `<button>` element.  (Same goes for "next" and "previous")

Reply:
I made this change. All the page items are now using the <Link> component.